### PR TITLE
Rename Rholang AST generated namespace & include BNFC generator in build compile step

### PR DESCRIPTION
**Pull request by tgrospic**
## Overview

This PR renames part of the package name of generated parser types from BNFC from _syntax_ to _ast_. The reason is to enable definition of extensions _syntax_ in Rholang project package file.

It replaces PR #3293.

Additionally build is modified to include in `compile` step also `bnfc:generate` which is not necessary to call manually. Also BNFC parser generated files are moved from `src` folder to `target`.

To rebuild the project from clean state it's now enough to do sbt `clean` and `compile`.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Review by zsluedem — APPROVED** [review]
LGTM

**Review by nzpr — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -77,120 +80,121 @@ $ docker logs -f rnode0
...
 Evaluating:
-5
+@{42}!("Hello!")
 ```
 
 A peer node can be started with the following command (note that `--bootstrap` takes the listening address of `rnode0`):
 
 ```bash
-$ docker run -v $HOME/tmp/peer1:/var/lib/rnode -it --name rnode1 --network rnode-net rchain/rnode:latest run --network localnet --bootstrap 'rnode://ee00a5357f2f4cb58b08a8a4c949da1b@172.18.0.2?protocol=40400&discovery=40404'
+$ docker run -it --rm --name rnode1 --network rnode-net rchain/rnode:latest run --bootstrap 'rnode://8c775b2143b731a225f039838998ef0fac34ba25@rnode0?protocol=40400&discovery=40404' --allow-private-addresses --host rnode1
```
Comment (zsluedem) [inline README.md:101]: Maybe we should create a wiki page for all the node addresses including testnet.

File: project/BNFC.scala [hunk 2]
```diff
@@ -89,25 +76,29 @@ object BNFC {
...
-        bnfcOutputDir  := (javaSource in Compile).value,
-        bnfcDocDir     := baseDirectory.value / "doc" / "bnfc",
-        clean          := cleanDir(bnfcOutputDir.value / nsToPath(bnfcNamespace.value)),
-        generate       := {
+        bnfcOutputDir  := (sourceManaged in Compile).value,
+        generate := {
           val fullCP = (fullClasspath in BNFCConfig).value
-          bnfcFiles(bnfcGrammarDir.value).foreach { (f: File) =>
-            bnfcGenerateSources(fullCP, f, bnfcOutputDir.value, bnfcNamespace.value)
+          val genCached = FileFunction.cached(
```
Comment (zsluedem) [inline project/BNFC.scala:84]: good one

**Comment by zsluedem** [comment]
good

File: README.md (hunk 1 above)
Comment (tgrospic) [reply README.md:101]: Agree, maybe even RCHIP as part of reference implementation and releases.

**Comment by tgrospic** [comment]
bors merge

**Pull request merged**